src/anyio/streams/memory.py

```diff
@@ -38,6 +38,12 @@ class MemoryObjectItemReceiver(Generic[T_Item]):
     task_info: TaskInfo = field(init=False, default_factory=get_current_task)
     item: T_Item = field(init=False)
 
+    def __repr__(self) -> str:
+        # When item is not defined, we get following error with default __repr__:
+        # AttributeError: 'MemoryObjectItemReceiver' object has no attribute 'item'
+        item = getattr(self, "item", None)
```

Comment:
I'm not sure that the item should appear at all in the `__repr__()`, but if it does, it should at least not appear if there isn't any.

Reply:
I can change as you wish, just please say how you would like it to be changed. 
The `repr` itself is not that important for me, the problem was for me that it failed with an error. 
And for debugging I used print and was really surprised when it failed.

The thing is, it sometimes works, but fails only when `item` is undefined.

For example, even **without** the fix code, I get something like this:
```
tool_stream_sender MemoryObjectSendStream(_state=MemoryObjectStreamState(max_buffer_size=inf, buffer=deque([]), open_send_channels=1, open_receive_channels=1, waiting_receivers=OrderedDict(), waiting_senders=OrderedDict()), _closed=False)
tool_stream_sender MemoryObjectSendStream(_state=MemoryObjectStreamState(max_buffer_size=inf, buffer=deque(['']), open_send_channels=1, open_receive_channels=1, waiting_receivers=OrderedDict(), waiting_senders=OrderedDict()), _closed=False)
```

And only then I get an error with traceback:
```
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "D:\Code\project\agents.py", line 250, in stream
    |     print("tool_stream_sender", repr(tool_stream_sender))
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "C:\Program Files\Python311\Lib\dataclasses.py", line 240, in wrapper
    |     result = user_function(self)
    |              ^^^^^^^^^^^^^^^^^^^
    |   File "<string>", line 3, in __repr__
    |   File "C:\Program Files\Python311\Lib\dataclasses.py", line 240, in wrapper
    |     result = user_function(self)
    |              ^^^^^^^^^^^^^^^^^^^
    |   File "<string>", line 3, in __repr__
    |   File "C:\Program Files\Python311\Lib\dataclasses.py", line 240, in wrapper
    |     result = user_function(self)
    |              ^^^^^^^^^^^^^^^^^^^
    |   File "<string>", line 3, in __repr__
    | AttributeError: 'MemoryObjectItemReceiver' object has no attribute 'item'
    +------------------------------------
```

So I just thought about the simplest solution, what I implemented

Reply:
By the way, thank you very much for the quick response!

Reply:
And I should add some tests for this, correct?
If yes, I will do it later today, I guess

Reply:
Yes, please.